lib/fog/storage/google_xml/real.rb

```diff
@@ -0,0 +1,109 @@
+module Fog
+  module Storage
+    class GoogleXML
+      class Real
+        include Utils
+
+        # Initialize connection to Google Storage
+        #
+        # ==== Notes
+        # options parameter must include values for :google_storage_access_key_id and
+        # :google_storage_secret_access_key in order to create a connection
+        #
+        # ==== Examples
+        #   google_storage = Storage.new(
+        #     :google_storage_access_key_id => your_google_storage_access_key_id,
+        #     :google_storage_secret_access_key => your_google_storage_secret_access_key
+        #   )
+        #
+        # ==== Parameters
+        # * options<~Hash> - config arguments for connection.  Defaults to {}.
+        #
+        # ==== Returns
+        # * Storage object with connection to google.
+        def initialize(options = {})
+          @google_storage_access_key_id = options[:google_storage_access_key_id]
+          @google_storage_secret_access_key = options[:google_storage_secret_access_key]
+          @connection_options = options[:connection_options] || {}
+          @hmac = Fog::HMAC.new("sha1", @google_storage_secret_access_key)
+          @host = options[:host] || "storage.googleapis.com"
+          @persistent = options.fetch(:persistent, true)
+          @port       = options[:port] || 443
+          @scheme     = options[:scheme] || "https"
+          @path_style = options[:path_style] || false
+        end
+
+        def reload
+          @connection.reset if @connection
+        end
+
+        def signature(params)
+          string_to_sign =
+            <<-DATA
```

Comment:
@plribeiro3000 
This now causes XML requests to fail because of whitespace in signature.
I would suggest either to rework it so it doesn't use DATA block or add `# rubocop:disable IndentAssignment` so it doesn't get picked up by Hound.